central/sensor/service/pipeline/networkflowupdate/flow_store_updater_impl.go

```diff
@@ -41,6 +43,31 @@ func (s *flowPersisterImpl) update(ctx context.Context, newFlows []*storage.Netw
 		s.firstUpdateSeen = true
 	}
 
+	if features.ExternalIPs.Enabled() {
+		// Sensor may have forwarded unknown NetworkEntities that we want to learn
+		for _, newFlow := range newFlows {
+			err := s.updateExternalNetworkEntityIfLearned(ctx, newFlow.GetProps().DstEntity)
+			if err != nil {
+				return err
+			}
+			err = s.updateExternalNetworkEntityIfLearned(ctx, newFlow.GetProps().SrcEntity)
+			if err != nil {
+				return err
+			}
+		}
+	} else {
+		// We are not storing the learned entities. Let net-flows point to INTERNET instead.
+		for newFlow := range flowsByIndicator {
+			if newFlow.SrcEntity.Learned {
```

Comment:
Should we use the getters here to protect us from nil issues or are we certain that `SrcEntity.Learned` will never be nil?  Same with `DstEntity`.

Reply:
I moved this new block of code to the beginning of the function in order to operate directly on the storage.NetworkEntities. Indeed, it is not possible to modify a map's key while iterating over it, which was what was happening.

Here is the new version of the code:

https://github.com/stackrox/stackrox/pull/11746/files#diff-3338cc43b98b5ad322cdd8cbd2ddb08e90f9a976f3c5cf1ec9247febe5d83dfaR42-R51